java/vector/src/main/java/org/apache/arrow/vector/VectorUnloader.java

```diff
@@ -80,7 +84,11 @@ private void appendNodes(FieldVector vector, List<ArrowFieldNode> nodes, List<Ar
     }
     buffers.addAll(fieldBuffers);
     for (FieldVector child : vector.getChildrenFromFields()) {
-      appendNodes(child, nodes, buffers);
+      if (child instanceof BinaryPointerVector) {
```

Comment:
it still seems like this code shoujldn't be needed, which method doesn't work inside append nodes that you need to special case this?